OOP Fundamentals and C#/09 - Design Patterns - Creational/Singleton/SafePrinter.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace Singleton
+{
+    public class SafePrinter
+    {
+        private static SafePrinter _instance;
+        private static readonly object padlock = new object();
+
+        private SafePrinter()
+        {
+        }
+
+        public static SafePrinter Instance
+        {
+            get
+            {
+                lock (padlock)
```

Comment:
I would suggest here to add one more if(instance == null)) before executing the lock. Lock is an expensive operation. and there's no need for it to be performed every time we need an instance. Maybe it is not really relevant for the topic of the lesson, but if we show this, I would say it's better to show it correctly